beacon-chain/blockchain/process_block.go

```diff
@@ -565,11 +569,16 @@ func (s *Service) handleBlockBLSToExecChanges(blk interfaces.BeaconBlock) error
 	if err != nil {
 		return errors.Wrap(err, "could not get BLSToExecutionChanges")
 	}
+	pool := s.cfg.BLSToExecPool
+	excessPending := pool.NumPending() - blsChangesPoolThreshold
 	for _, change := range changes {
 		if err := s.cfg.BLSToExecPool.MarkIncluded(change); err != nil {
 			return errors.Wrap(err, "could not mark BLSToExecutionChange as included")
 		}
 	}
+	if excessPending > 0 && len(changes) > excessPending {
```

Comment:
I wonder if its worth keeping track of how many items the map has held over its lifetime to get a rough idea of how large the backing arrays are, just to avoid cycling through maps (which is quite expensive, copying over all the elements) when there isn't really much of a benefit. Too bad we can't check the cap of a map :( https://github.com/golang/go/issues/52157

Reply:
It seems like this is cluttering up the implementation of `handleBlockBLSToExecChanges`, could we move it into the internals of the pool manager?

Reply:
`MarkIncluded` would be a good place for it.